includes/class-gp-notifications.php

```diff
@@ -0,0 +1,463 @@
+<?php
+/**
+ * Routes: GP_Notifications class
+ *
+ * Manages the notifications of the plugin.
+ *
+ * @package gp-translation-helpers
+ * @since 0.0.2
+ */
+class GP_Notifications {
+	/**
+	 * Sends notifications when a new comment in the discussion is stored using the WP REST API.
+	 *
+	 * @since 0.0.2
+	 *
+	 * @param WP_Comment      $comment  Inserted or updated comment object.
+	 * @param WP_REST_Request $request  Request object.
+	 * @param bool            $creating True when creating a comment, false when updating.
+	 *
+	 * @return void
+	 */
+	public static function init( WP_Comment $comment, $request, $creating ) {
+		$post = get_post( $comment->comment_post_ID );
+		if ( Helper_Translation_Discussion::POST_TYPE === $post->post_type ) {
+			if ( ( '1' === $comment->comment_approved ) || ( 'approve' === $comment->comment_approved ) ) {
+				$comment_meta = get_comment_meta( $comment->comment_ID );
+				if ( ( '0' !== $comment->comment_parent ) ) { // Notify to the thread only if the comment is in a thread.
+					self::send_emails_to_thread_commenters( $comment, $comment_meta );
+				}
+				$root_comment      = self::get_root_comment_in_a_thread( $comment );
+				$root_comment_meta = get_comment_meta( $root_comment->comment_ID );
+				if ( array_key_exists( 'comment_topic', $root_comment_meta ) ) {
+					switch ( $root_comment_meta['comment_topic'][0] ) {
+						case 'typo':
+						case 'context': // Notify to the GlotPress admins.
+							self::send_emails_to_gp_admins( $comment, $comment_meta );
+							break;
+						case 'question': // Notify to the project validator.
+							self::send_emails_to_validators( $comment, $comment_meta );
+							break;
+					}
+				}
+			}
+		}
+	}
+
+	/**
+	 * Sends notifications when a comment changes its status to "approved".
+	 *
+	 * @since 0.0.2
+	 *
+	 * @param int|string $new_status The new comment status.
+	 * @param int|string $old_status The old comment status.
+	 * @param WP_Comment $comment    The comment object.
+	 *
+	 * @return void
+	 */
+	public static function comment_change_status( $new_status, $old_status, WP_Comment $comment ) {
```

Comment:
Could we rename the function to make it more clear that this is called in a hook?

```suggestion
	public static function on_comment_status_change( $new_status, $old_status, WP_Comment $comment ) {
```